testing_and_setup/compass/ocean/global_ocean/ARM60to10/init/define_base_mesh.py

```diff
@@ -1,41 +1,131 @@
-
     AtlNH = RRS30to10
-    AtlGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
-
+    AtlVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, AtlNH, 0, 6)
     PacNH = mdt.mergeCellWidthVsLat(lat, 30 * QU1, RRS30to10, 50, 10)
-    PacGrid = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+    PacVsLat = mdt.mergeCellWidthVsLat(lat, EC60to30, PacNH, 0, 6)
+
+    # Expand from 1D to 2D
+    _, AtlGrid = np.meshgrid(lon, AtlVsLat)
+    _, PacGrid = np.meshgrid(lon, PacVsLat)
+
+    # Signed distance of Atlantic region
+    fc = read_feature_collection('Atlantic_region.geojson')
+    signedDistance = signed_distance_from_geojson(fc, lon, lat,
+                                                 max_length=0.25)
+
+    # Merge Atlantic and Pacific distrubutions smoothly
+    transitionWidth = 500.0e3  # [m]
+    maskSmooth = 0.5 * (1 + np.tanh(signedDistance / transitionWidth))
+    cellWidthSmooth = PacGrid * maskSmooth + AtlGrid * (1 - maskSmooth)
 
-    cellWidth = mdt.AtlanticPacificGrid(lat, lon, AtlGrid, PacGrid)
+    # Merge Atlantic and Pacific distrubutions with step function
+    maskSharp = 0.5 * (1 + np.sign(signedDistance))
+    cellWidthSharp = PacGrid * maskSharp + AtlGrid * (1 - maskSharp)
 
-    import matplotlib.pyplot as plt
-    import matplotlib
-    matplotlib.use('Agg')
+    # Create a land mask that is 1 over land
+    fc = read_feature_collection('Americas_land_mask.geojson')
+    Americas_land_mask = mask_from_geojson(fc, lon, lat)
+    fc = read_feature_collection('Europe_Africa_land_mask.geojson')
+    Europe_Africa_land_mask = mask_from_geojson(fc, lon, lat)
+    landMask = np.fmax(Americas_land_mask, Europe_Africa_land_mask)
+
+    # Merge: step transition over land, smooth transition over water
+    cellWidth = cellWidthSharp * landMask + cellWidthSmooth * (1 - landMask)
+
+    # save signed distance to a file
+    # da = xarray.DataArray(signedDistance,
+    #                      dims=['y', 'x'],
+    #                      coords={'y': lat, 'x': lon},
+    #                      name='signedDistance')
+    #cw_filename = 'signedDistance.nc'
+    # da.to_netcdf(cw_filename)
+
+    print('plotting ...')
+    fig = plt.figure()
     plt.clf()
-    plt.plot(lat, AtlGrid, label='Atlantic')
-    plt.plot(lat, PacGrid, label='Pacific')
-    plt.grid(True)
-    plt.xlabel('latitude')
-    plt.title('Grid cell size, km')
+    fig.set_size_inches(10.0, 10.0)
+
+    ax = plt.subplot(4, 2, 1)
+    ax.plot(lat, AtlVsLat, label='Atlantic')
+    ax.plot(lat, PacVsLat, label='Pacific')
+    ax.grid(True)
+    plt.title('Grid cell size [km] versus latitude')
     plt.legend()
-    plt.savefig('cellWidthVsLat.png')
+
+    varNames = [
+        'signedDistance',
+        'maskSmooth',
+        'cellWidthSmooth',
+        'maskSharp',
+        'cellWidthSharp',
+        'landMask',
+        'cellWidth']
+    j = 2
+    for varName in varNames:
+        plot_cartopy(j, varName, vars()[varName])
```

Comment:
I've never used `vars()` to get the dictionary of local variables like this but it seems like a clever approach to me.